app/ports/utilities/sort_by_version.py

```diff
@@ -0,0 +1,6 @@
+def sort_dicts_by_version(dictionary, key):
+    return sorted(dictionary, key=lambda x: tuple(int(i) for i in x[key].split(".")), reverse=True)
```

Comment:
I believe the function name, as well as the argument name, are a little misleading as you are actually not sorting the dictionary here. 
I'd also take out the `tuple(int(i) for i in x[key].split("."))` part outside.
```
def split_version(a):
    return tuple(int(i) for i in a.split("."))

def sort_list_by_version(lst):
    return sorted(lst, key=split_version, reverse=True)
...
```
That being said, I'd actually look into writing a version comparator of our own or maybe find an implementation online somewhere in the near future.
That would totally help to avoid splitting the os_version again where you need to set the `cxx_stdlib`'s default value when the os_versions >= 10.9 or many other places perhaps.

Reply:
Thank you.

Reply:
> ```
> def split_version(a):
>     return tuple(int(i) for i in a.split("."))
> 
> def sort_list_by_version(lst):
>     return sorted(lst, key=split_version, reverse=True)
> ...
> ```

This won't work probably, and I am getting an error. It seems that `key` takes in a sorted list. Also, we have to sort "list" and "list of dictionaries" by some specific key. So, I believe we will need two such functions.

Reply:
> That being said, I'd actually look into writing a version comparator of our own or maybe find an implementation online somewhere in the near future.

I think distutils.version should work well enough, and is in the python standard library.

Reply:
While it's still not in a state what I had in mind, this certainly looks better.